tket/src/Circuit/macro_circ_info.cpp

```diff
@@ -143,9 +143,7 @@ Circuit Circuit::subcircuit(const Subcircuit& sc) const {
       port_t in_port = get_source_port(e);
       OpType type = sub.get_OpType_from_Vertex(sub_source);
       if (is_initial_q_type(type) || type == OpType::ClInput) {
-        // For Quantum and Classical edges, boundary_edge == *it;
-        // for Boolean, this gives the corresponding Classical
-        Edge boundary_edge = this->get_nth_out_edge(source, in_port);
+        Edge boundary_edge = get_linear_edge(e);
```

Comment:
What's wrong with the previous solution?

Reply:
Nothing, does the same thing. But I needed the same code elsewhere so I factored it out (and removed the comment because it was redundant).